z390test/src/test/groovy/org/z390/test/RunTestOption.groovy

```diff
@@ -0,0 +1,109 @@
+package org.z390.test
+
+import org.junit.jupiter.api.Test
+
+class RunTestOption extends z390Test {
+
+    @Test
+    void test_TESTOPT_OK() {
+        /**
+         * test 1 - should assemble okay
+         */
+        int rc = this.asm(basePath("rt", "mlc", "TESTOPT"), "@${basePath("rt", "opt", "testopt0.OPT")}")
+        this.printOutput()
+        assert rc == 0
+    }
+
+    @Test
+    void test_TESTOPT1_invalidOption() {
+        /**
+         * test 2 - should assemble with error for invalid option
+         */
+        int rc = this.asm(basePath("rt", "mlc", "TESTOPT"), "@${basePath("rt", "opt", "testopt1.OPT")}")
+        this.printOutput()
+        assert rc == 16
+        assert this.fileData['ERR'] =~ /invalid options -\s{2}chksrc\(2X\)/
+        assert this.stdout =~ /invalid options -\s{2}chksrc\(2X\)/
+    }
+
+    @Test
+    void test_TESTOPT2L_validReference() {
+        /**
+         * test 3 - should assemble okay with valid reference
+         */
+        var testOpt0FileContents = """
+* Test with a valid option
+  chksrc(2)  * Check sources for illegal characters
+"""
+        var innerOptionFilename = createTempFile("testopt0.OPT", testOpt0FileContents)
+        var testOpt2LFileContents = """
+* Test with a valid subfile
+  chksrc(2)  * Check sources for illegal characters
+@${innerOptionFilename}
+"""
+        var OptionFilename = createTempFile("testopt2L.OPT", testOpt2LFileContents)
+        int rc = this.asm(basePath("rt", "mlc", "TESTOPT"), "@${OptionFilename}")
+        this.printOutput()
+        assert rc == 0
+        assert fileData['PRN'] =~ /testopt2L.OPT=/
+        assert fileData['PRN'] =~ /testopt0.OPT=/
+    }
+
+    @Test
+    void test_TESTOPT3L_incorrectReference() {
+        /**
+         * test 4 - should assemble with error for incorrect reference
+         */
+        int rc = this.asm(basePath("rt", "mlc", "TESTOPT"), "@${basePath("rt", "opt", "testopt3L.OPT")}")
+        this.printOutput()
+        assert rc == 16
+        assert this.stdout =~ /TZ390E abort error 21 - invalid options -\s{2}@rt\/opt\/testoptX/
+        assert this.fileData['ERR'] =~ /TZ390E abort error 21 - invalid options -\s{2}@rt\/opt\/testoptX/
```

Comment:
Not sure when testing on Windows what z390 does with the Linux path separator characters in testopt3L.opt. In any case, Windows or Linux/Mac, it does contain an invalid file name, so okay.

Reply:
The test checks that the error message matches.. I have run the tests on both Linux and Windows.. In this case, looks like a hardcoded backslash (\) is used, but as you say the test is an invalid filename so may not matter.